### DISCUSSION TITLE
Is ZAP crossplatform?

### DISCUSSION THREAD

**Discussion by OsakiTsukiko** [zigzap/zap#21] [General]
I'm not very familiar with facil.io, but from my understanding, it only runs on Linux, BSD, and macOS. Is this the case for ZAP as well or did I misunderstand something?

Answer: Kinda..

**Comment by renerocksai** [comment]
Hi, I have no hard evidence for this but my assumptions are as follows, based on the evidence:

- facil.io defines its subset of platforms
- zig itself can build on a huge set of platforms

Therefore I would assume that zap works on all platform that are supported by facil.io
 
If, for instance, you want to run it on Windows (which, IMHO, is a bad idea for production in general and IF, then maybe acceptable for development) then I'd recommend trying to run it on the WSL2 layer. I am not sure how well emulation of e.g. epoll works on Windows, so, again, it might be OK for development, but I wouldn't expect exceptional performance on Windows.

**Comment by renerocksai** [comment]
Then again, hardcore Windows systems developers might have more insights into how well WSL2 can run code like facil.io.

**Comment by OsakiTsukiko** [comment]
Yea, my main concern was, as you guessed, development. If I use an os locked lib in an open source project that might cut off a big portion of 'might be' contributors. And while WSL is def a valid option, it might be too much of a hassle for some. Thx for the quick response tho
